app/src/main/kotlin/com/wire/android/ui/userprofile/other/OtherUserProfileScreenViewModel.kt

```diff
@@ -210,6 +219,43 @@ class OtherUserProfileScreenViewModel @Inject constructor(
         _infoMessage.emit(type.uiText)
     }
 
+    fun openRemoveConversationMemberDialog() {
+        viewModelScope.launch {
+            removeConversationMemberDialogState = DialogState.Visible(
+                PreservedState.State(
+                    RemoveConversationMemberState(
+                        conversationId = conversationId!!,
+                        fullName = state.fullName,
+                        userName = state.userName,
+                        userId = userId
+                    )
+                )
+            )
+        }
+    }
+
+    fun hideRemoveConversationMemberDialog() {
+        viewModelScope.launch {
+            removeConversationMemberDialogState = DialogState.Hidden
+        }
+    }
+
+    fun removeConversationMember(preservedState: PreservedState<RemoveConversationMemberState>) {
+        viewModelScope.launch {
+            removeConversationMemberDialogState = DialogState.Visible(
+                preservedState.toLoading()
+            )
+            when (removeMemberFromConversation(state.groupState!!.conversationId, userId)) {
+                is RemoveMemberFromConversationUseCase.Result.Failure -> {
+                    showInfoMessage(InfoMessageType.RemoveConversationMemberError)
+                    removeConversationMemberDialogState = DialogState.Hidden
+                }
+                RemoveMemberFromConversationUseCase.Result.Success -> removeConversationMemberDialogState =
+                    DialogState.Hidden
+            }
```

Comment:
Hiding the dialog is duplicated so maybe:

```suggestion
            if (removeMemberFromConversation(state.groupState!!.conversationId, userId) is RemoveMemberFromConversationUseCase.Result.Failure)
                showInfoMessage(InfoMessageType.RemoveConversationMemberError)
            removeConversationMemberDialogState = DialogState.Hidden
```

Reply:
done